### PR TITLE
Add git pre-commit hook

### PR DESCRIPTION
**Pull request by fhunleth**

### CODE REVIEW

**Review by mattludwigs — APPROVED** [review]

**Comment by fhunleth** [comment]
I'm not 100% sure how I feel about pre-commit hooks and committing the scripts to repos. I "think" this help me more than others since you have to opt in to use the hook, and I wouldn't expect most people to opt into pre-commit hooks unless they were contributing a lot to this project.

File: pre-commit.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+#
+# git pre-commit hook
+#
+# Install by:
+# $ ln -s ../../pre-commit.sh .git/hooks/pre-commit
```
Comment (mattludwigs) [inline pre-commit.sh:7]: Should this be `ln -s pre-commit.sh .git/hooks/pre-commit`?

Comment (fhunleth) [reply pre-commit.sh:7]: The `../..` is to go back from `.git/hooks` to get to the base directory.

Comment (mattludwigs) [reply pre-commit.sh:7]: Okay, I get it. :+1:

**Review by ConnorRigby — APPROVED** [review]

**Pull request merged**